From a8cfda743b23c30ba2901ded086e2aeb53411e66 Mon Sep 17 00:00:00 2001
From: Ranjan Dutta <ranjan.dutta@intel.com>
Date: Tue, 13 Apr 2021 11:32:16 +0800
Subject: [PATCH 5/5] Revert "drm/i915/gvt: Set SNOOP for PAT3 on BXT/APL to
 workaround GPU BB hang"

This reverts commit 8cd68991b836feb73dccf72a9ffa222ba4c6dab4.
---
 drivers/gpu/drm/i915/gvt/handlers.c | 32 +----------------------------
 1 file changed, 1 insertion(+), 31 deletions(-)

diff --git a/drivers/gpu/drm/i915/gvt/handlers.c b/drivers/gpu/drm/i915/gvt/handlers.c
index f14aefcb5b7b..689b07bc91c4 100644
--- a/drivers/gpu/drm/i915/gvt/handlers.c
+++ b/drivers/gpu/drm/i915/gvt/handlers.c
@@ -1632,34 +1632,6 @@ static int edp_psr_imr_iir_write(struct intel_vgpu *vgpu,
 	return 0;
 }
 
-/**
- * FixMe:
- * If guest fills non-priv batch buffer on ApolloLake/Broxton as Mesa i965 did:
- * 717e7539124d (i965: Use a WC map and memcpy for the batch instead of pwrite.)
- * Due to the missing flush of bb filled by VM vCPU, host GPU hangs on executing
- * these MI_BATCH_BUFFER.
- * Temporarily workaround this by setting SNOOP bit for PAT3 used by PPGTT
- * PML4 PTE: PAT(0) PCD(1) PWT(1).
- * The performance is still expected to be low, will need further improvement.
- */
-static int bxt_ppat_low_write(struct intel_vgpu *vgpu, unsigned int offset,
-			      void *p_data, unsigned int bytes)
-{
-	u64 pat =
-		GEN8_PPAT(0, CHV_PPAT_SNOOP) |
-		GEN8_PPAT(1, 0) |
-		GEN8_PPAT(2, 0) |
-		GEN8_PPAT(3, CHV_PPAT_SNOOP) |
-		GEN8_PPAT(4, CHV_PPAT_SNOOP) |
-		GEN8_PPAT(5, CHV_PPAT_SNOOP) |
-		GEN8_PPAT(6, CHV_PPAT_SNOOP) |
-		GEN8_PPAT(7, CHV_PPAT_SNOOP);
-
-	vgpu_vreg(vgpu, offset) = lower_32_bits(pat);
-
-	return 0;
-}
-
 static int mmio_read_from_hw(struct intel_vgpu *vgpu,
 		unsigned int offset, void *p_data, unsigned int bytes)
 {
@@ -2806,7 +2778,7 @@ static int init_broadwell_mmio_info(struct intel_gvt *gvt)
 
 	MMIO_DH(GEN6_PCODE_MAILBOX, D_BDW_PLUS, NULL, mailbox_write);
 
-	MMIO_D(GEN8_PRIVATE_PAT_LO, D_BDW_PLUS & ~D_BXT);
+	MMIO_D(GEN8_PRIVATE_PAT_LO, D_BDW_PLUS);
 	MMIO_D(GEN8_PRIVATE_PAT_HI, D_BDW_PLUS);
 
 	MMIO_D(GAMTARBMODE, D_BDW_PLUS);
@@ -3309,8 +3281,6 @@ static int init_bxt_mmio_info(struct intel_gvt *gvt)
 
 	MMIO_DFH(GEN9_CTX_PREEMPT_REG, D_BXT, F_CMD_ACCESS, NULL, NULL);
 
-	MMIO_DH(GEN8_PRIVATE_PAT_LO, D_BXT, NULL, bxt_ppat_low_write);
-
 	return 0;
 }
 
-- 
2.25.1

